inst/scribble/scribble.js

```diff
@@ -0,0 +1,626 @@
+
+    const scaleRatio = parseFloat(
+      this.getVisibleSlideScaleRatio().match('\\d+.\\d+')
+    )
+
+    // Resize canvas itself
+    this.fabrics.forEach((fab) => {
+      fab.setWidth(scalerSize.width)
+      fab.setHeight(scalerSize.height)
+      fab.setZoom(scaleRatio)
+    })
+
+    // Reposition toolbox outside slides if there is extra space
+    if ((window.innerWidth - scalerSize.width) / 2 >= 42) {
+      this.toolBox.style.right = '-40px'
+    } else {
+      this.toolBox.style.right = '10px'
+    }
+  }
+
+  createButton (id, name, title) {
+    const btn = document.createElement('button')
+    btn.classList.add('xe-scribble__button', 'xe-scribble__button__' + id)
+    btn.innerHTML = this.svgs[name]
+    btn.title = title || (name.slice(0, 1).toUpperCase() + name.slice(1))
+    return btn
+  }
+
+  createColorPicker () {
+    const colorPicker = document.createElement('input')
+    colorPicker.setAttribute('id', 'colorPicker')
+    colorPicker.classList.add('xe-scribble__button')
+    colorPicker.setAttribute('type', 'color')
+    colorPicker.setAttribute('value', this.currColor)
+    colorPicker.title = 'Choose Pen Color'
+    return colorPicker
+  }
+
+  createToolbox () {
+    this.toolBox = document.createElement('div')
+    this.toolBox.classList = 'xe-scribble__tools'
+  }
+
+  rgbColorToHex (color) {
+    // https://haacked.com/archive/2009/12/29/convert-rgb-to-hex.aspx/
+    if (color.substr(0, 1) === '#') {
+      return color
+    }
+    const digits = /rgb\((\d+), (\d+), (\d+)\)/.exec(color)
+
+    const red = parseInt(digits[1])
+    const green = parseInt(digits[2])
+    const blue = parseInt(digits[3])
+
+    const rgb = blue | (green << 8) | (red << 16)
+    return '#' + rgb.toString(16).padStart(6, '0')
+  }
+
+  pickContrastForegroundColor (color, light, dark, threshold) {
+    color = this.rgbColorToHex(color)
+
+    // https://stackoverflow.com/a/41491220/2022615
+    light = light || '#FFFFFF'
+    dark = dark || '#000000'
+    threshold = threshold || 145
+    color = (color.charAt(0) === '#') ? color.substring(1, 7) : color
+    const r = parseInt(color.substring(0, 2), 16) // hexToR
+    const g = parseInt(color.substring(2, 4), 16) // hexToG
+    const b = parseInt(color.substring(4, 6), 16) // hexToB
+    const score = ((r * 0.299) + (g * 0.587) + (b * 0.114))
+    return (score > threshold) ? dark : light
+  }
+
+  assembleToolbox () {
+    this.createToolbox()
+
+    // Build draw, erase, clear buttons
+    this.drawBtn = this.createButton('draw', 'draw', 'Start Drawing (S)')
+    this.eraseBtn = this.createButton('erase', 'eraser', 'Erase Lines')
+    this.clearBtn = this.createButton('clear', 'trash', 'Clear This Slide')
+
+    this.colorPicker = this.createColorPicker()
+    this.colorPicker.addEventListener('input', () => {
+      this.currColor = this.colorPicker.value
+      this.currFabric.freeDrawingBrush.color = this.currColor
+      this.setPenColorCSSVariables(this.currColor)
+    })
```

Comment:
I just noticed a bug. When you click the eraser, the eraser path color is set to `this.transparent`, but, if you make an input to the color picker while in eraser mode, the code above will change the transparent eraser paths to the newly chosen color, and will draw lines of this color while "erasing".

I think we just need a conditional statement in the above code to avoid changing the brush color while in eraser mode, something like:
```js
this.colorPicker.addEventListener('input', () => {
      this.currColor = this.colorPicker.value
      if (this.drawMode) this.currFabric.freeDrawingBrush.color = this.currColor
      this.setPenColorCSSVariables(this.currColor)
    })
```

Reply:
Good catch, thanks!